dcore/forcing.py

```diff
@@ -0,0 +1,73 @@
+from abc import ABCMeta, abstractmethod
+
+class Forcing(object):
+    """
+    Base class for forcing terms for dcore.
+
+    :arg state: x :class:`.State` object.
+    """
+    __metaclass__ = ABCMeta
+
+    def __init__(self, state):
+        self.state = state
+    
+    @abstractmethod
+    def apply(self,scale,):
+        """
+        Function takes x as input, computes F(x) and returns 
+        x_out = x + scale*F(x) 
+        as output.
+
+        :arg scale: parameter to scale the output by.
+        :arg x: :class:`.Function` object, the input Function.
+        :arg x_out: :class:`.Function` object, the output Function.
+        """
+        pass
```

Comment:
If the forcing is always, compute f(x), and add it with some scaling, can the implementation of `apply` not live here?